docs/pages/upgrading/reference.mdx

```diff
@@ -0,0 +1,502 @@
+---
+title: Upgrading Reference
+description: Provides detailed information on upgrading Teleport in various situations.
+tocDepth: 3
+---
+
+This guide explains how to upgrade components of a Teleport cluster in
+non-standard situations. 
```

Comment:
I'm not sure why self-hosted users trying to setup auto-upgrade is a non-standard situation. Maybe rephrase this to say this is targeting self-hosted setups?